src/erasure-code/jerasure/ErasureCodeJerasure.cc

```diff
@@ -141,13 +141,15 @@ void ErasureCodeJerasure::encode_delta(const bufferptr &old_data,
                                        const bufferptr &new_data,
                                        bufferptr *delta)
 {
-  ceph_assert(false); // Not yet supported by this plugin
-}
+  //TODO handle old_data or new_data being a special zero buffer.
+  for (unsigned int i = 0; i < delta->length(); i++) {
```

Comment:
This looks like a badly written memcpy! Given that the interface permits old_data and delta to be the same bufferptr you should check this and avoid performing any copy if they are the same, otherwise perform a memcpy.